core/vm/message/message_util.cpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "vm/message/message_util.hpp"
+#include "storage/hamt/hamt.hpp"
+
+namespace fc::vm::message {
+
+  using common::getCidOf;
+
+  outcome::result<CID> cid(const UnsignedMessage &m) {
+    auto encoded = serialize<UnsignedMessage>(m);
+    return getCidOf(encoded);
+  }
+
+  outcome::result<CID> cid(const SignedMessage &sm) {
```

Comment:
cid(bls_signed) == cid(unsigned)? And different for secp256k1? Is it correct?

Reply:
yes, at least as long as the Lotus implementation can be trusted